sw/airborne/firmwares/rotorcraft/stabilization/stabilization_indi.c

```diff
@@ -35,139 +35,149 @@
-  },
-
-#if STABILIZATION_INDI_USE_ADAPTIVE
-  .adaptive = TRUE,
-#else
-  .adaptive = FALSE,
-#endif
-};
 
 #if PERIODIC_TELEMETRY
 #include "subsystems/datalink/telemetry.h"
 
 static void send_att_indi(struct transport_tx *trans, struct link_device *dev)
 {
-  //The estimated G values are scaled, so scale them back before sending
-  struct FloatRates g1_disp;
-  RATES_SMUL(g1_disp, indi.est.g1, INDI_EST_SCALE);
-  float g2_disp = indi.est.g2 * INDI_EST_SCALE;
-
   pprz_msg_send_STAB_ATTITUDE_INDI(trans, dev, AC_ID,
-                                   &indi.rate.dx.p,
-                                   &indi.rate.dx.q,
-                                   &indi.rate.dx.r,
-                                   &indi.angular_accel_ref.p,
-                                   &indi.angular_accel_ref.q,
-                                   &indi.angular_accel_ref.r,
-                                   &g1_disp.p,
-                                   &g1_disp.q,
-                                   &g1_disp.r,
-                                   &g2_disp);
+                                   &G1[1][0],
+                                   &G1[1][1],
+                                   &G1[1][2],
+                                   &G1[1][3],
+                                   &G1[2][0],
+                                   &G1[2][1],
+                                   &G1[2][2],
+                                   &G1[2][3],
+                                   &G1[3][0],
+                                   &G1[3][1]);
 }
 #endif
 
 void stabilization_indi_init(void)
 {
-  // Initialize filters
-  stabilization_indi_second_order_filter_init(&indi.rate, STABILIZATION_INDI_FILT_OMEGA, STABILIZATION_INDI_FILT_ZETA, STABILIZATION_INDI_FILT_OMEGA_R);
-  stabilization_indi_second_order_filter_init(&indi.u, STABILIZATION_INDI_FILT_OMEGA, STABILIZATION_INDI_FILT_ZETA, STABILIZATION_INDI_FILT_OMEGA_R);
-  stabilization_indi_second_order_filter_init(&indi.est.rate, 10.0, 0.8, 10.0); //FIXME: no magic number
-  stabilization_indi_second_order_filter_init(&indi.est.u, 10.0, 0.8, 10.0); //FIXME: no magic number
-
+  stabilization_indi_second_order_filter_init(&indi_rate, STABILIZATION_INDI_FILT_OMEGA, STABILIZATION_INDI_FILT_ZETA, STABILIZATION_INDI_FILT_OMEGA_R, 3);
+  stabilization_indi_second_order_filter_init(&actuator_state_filt, STABILIZATION_INDI_FILT_OMEGA, STABILIZATION_INDI_FILT_ZETA, STABILIZATION_INDI_FILT_OMEGA_R, ACTUATORS_NB);
+  stabilization_indi_second_order_filter_init(&estimation_input, IDENTIFICATION_INDI_FILT_OMEGA, IDENTIFICATION_FILT_ZETA, IDENTIFICATION_INDI_FILT_OMEGA, ACTUATORS_NB);
+  stabilization_indi_second_order_filter_init(&estimation_output, IDENTIFICATION_INDI_FILT_OMEGA, IDENTIFICATION_FILT_ZETA, IDENTIFICATION_INDI_FILT_OMEGA, 3);
 #if PERIODIC_TELEMETRY
   register_periodic_telemetry(DefaultPeriodic, PPRZ_MSG_ID_STAB_ATTITUDE_INDI, send_att_indi);
 #endif
 }
 
+#define VECT_ASSIGN(_v, _b, _n) {  \
```

Comment:
we have at least for float vectors: https://github.com/paparazzi/paparazzi/blob/master/sw/airborne/math/pprz_algebra_float.h#L501